rfcs/text/0000-application-monitoring-for-ga.md

```diff
@@ -0,0 +1,126 @@
+| Metrics for AWS applications and managed services | [CloudWatch Metrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Metric) | [console](https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#metricsV2:) | adopt as best practice for AWS |
+| Metrics for GCP applications and managed services | [Stackdriver Metrics](https://cloud.google.com/monitoring/api/metrics) | [console](https://console.cloud.google.com/monitoring) | adopt as best practice for GCP |
+| Alerting based on CloudWatch Metrics | [CloudWatch Alarms](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#CloudWatchAlarms) | [github+terraform](https://github.com/HumanCellAtlas/dcp-monitoring) | adopt as best practice |
+| Metrics based on operational log indices | [Elasticsearch Metrics](http://docs.grafana.org/features/datasources/elasticsearch/) | [dev/integration/staging](https://logs.dev.data.humancellatlas.org/), [prod](https://logs.data.humancellatlas.org/) | use not recommended, but sometimes necessary |
+| Build statuses, system health, and availability reporting | [Status API](https://github.com/HumanCellAtlas/status-api) | [dev/integration/staging](https://status.dev.data.humancellatlas.org/), [prod](https://status.data.humancellatlas.org/) | adopt as best practice |
+| Public presentation of build statuses, system health, and availability reporting | [DCP Status Page](https://github.com/HumanCellAtlas/humancellatlas.github.io) | [https://humancellatlas.github.io/](https://humancellatlas.github.io/) | temporary until formal status page is built |
+
+### Black-box monitoring
+
+Black-box monitoring attempts to test system behavior as a user would see it.
+
+To estimate and check this behavior, implement an HTTP health check endpoint for your service and configure a [Route53 health check](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/route-53-concepts.html#route-53-concepts-health-check) in the [dcp-monitoring](https://github.com/HumanCellAtlas/dcp-monitoring) repository.
+
+Your system's health check should aggregate the health status of its API and any infrastructure it depends on. For example, if your service is a HTTP REST API that holds state in a database, the health check endpoint would return a healthy status if the API was working and a basic query, such as `SELECT 1`, could be performed against the database and an unhealthy status otherwise.
+
+### White-box monitoring
+
+In order to diagnose problems, critical application logic and infrastructure that each system depends on must be monitored. To illustrate, consider an API that depends on AWS S3 and Elasticsearch. If the API goes down and its logs are not informative, there is no way of telling what infrastructure component failed unless there is separate monitoring of each component.
+
```

Comment:
If uninformative logs are the problem, it seems that improving the logging would be more productive approach than providing a different mechanism.  Logging is often white-box, going deeper than just the HTTP logging.  One thing to consider is white-box debugging.  Being able to connect to the node running a given service, examine the state of the system ans start a debugger against a running process can be very helpful with unexpected problems that don't result it a stack trace or similar.

Reply:
Good point. Changing the language here.